### PR TITLE
Setup P2P and Peer structure for lisk-elements@lisk-p2p - Closes#890

### LINKED ISSUE #890: Create package and setup boilerplate for lisk-p2p
### Description
In order to start developing the lisk-p2p, we need a basic setup similar to other packages. Setup basic structure for Peer and P2P functionality along with unit tests.

### PR DESCRIPTION
**Pull request by ishantiw**
### Description

In order to start developing the lisk-p2p, we need a basic setup similar to other packages.
Setup P2P and Peer structure for lisk-elements@lisk-p2p. Add related classes and interfaces with unit tests.
- Add Peer and P2P class skelton with required interfaces
- Create unit tests
- Comply with lint rules

### How to test it?
`npm run lint`
`npm test` or
`npm run build` and try to use each pakcages

### Review checklist

* The PR resolves #890 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-p2p/src/blockchain/blockchain_p2p.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+interface IHistogram {
+	[key: number]: number;
+}
+interface IHistogramValues {
+	height: number;
+	histogram: IHistogram;
+	max: number;
+}
+export class BlockchainP2P {
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply packages/lisk-p2p/src/blockchain/blockchain_p2p.ts]: Should we remove `blockchain` prefix from the file name as well as we are not using generic classes? same goes for `blockchain_peer.ts` and `peer.ts`

File: packages/lisk-p2p/src/blockchain/blockchain_p2p.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	// TODO
+	public request = async (
+		packet: IP2PRequestPacket,
+	): Promise<IP2PResponsePacket> => {
+		const response = packet;
+
+		return Promise.resolve(response);
+	};
+
+	public selectPeers = (
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply packages/lisk-p2p/src/blockchain/blockchain_p2p.ts]: I will create a new PR

File: packages/lisk-p2p/src/blockchain/blockchain_p2p.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (shuse2) [reply packages/lisk-p2p/src/blockchain/blockchain_p2p.ts]: I think it would be better as our p2p main focus is blockchain anyway =)

**Comment by ishantiw** [comment]
@shuse2 Addressed

**Pull request merged**